src/langchain_google_firestore/vectorstores.py

```diff
@@ -0,0 +1,354 @@
+
+        for batch in more_itertools.chunked(texts, WRITE_BATCH_SIZE):
+            texts_embs = self.embedding.embed_documents(batch)
+            for i, text in enumerate(batch):
+                doc_id = ids[i] if ids else None
+                doc = self.collection.document(doc_id)
+                _ids.append(doc.id)
+
+                data = {
+                    self.content_field: text,
+                    self.embedding_field: Vector(texts_embs[i]),
+                }
+
+                if metadatas:
+                    data[self.metadata_field] = metadatas[i]
+
+                db_batch.set(doc, data, merge=True)
+
+            db_batch.commit()
+
+        return _ids
+
+    def delete(self, ids: Optional[List[str]] = None, **kwargs: Any) -> None:
+        """Delete documents from the vector store.
+
+        Args:
+            ids: The document ids to delete from the vector store.
+        """
+
+        if not ids or len(ids) == 0:
+            return
+
+        for batch in more_itertools.chunked(ids, WRITE_BATCH_SIZE):
+            db_batch = self.client.batch()
+            for doc_id in batch:
+                doc_ref = self.collection.document(doc_id)
+                db_batch.delete(doc_ref)
+            db_batch.commit()
+
+    def _similarity_search(
+        self, query: List[float], k: int = DEFAULT_TOP_K, **kwargs: Any
+    ) -> List[DocumentSnapshot]:
+        filters = kwargs.get("filters") or self.filters
+
+        wfilters = None
+
+        if filters is not None:
+            wfilters = self.collection.where(filter=filters)
+
+        results = (wfilters or self.collection).find_nearest(
+            vector_field=self.embedding_field,
+            query_vector=Vector(query),
+            distance_measure=self.distance_strategy,
+            limit=k,
+        )
+
+        return results.get()
+
+    def similarity_search(
+        self,
+        query: str,
+        k: int = DEFAULT_TOP_K,
+        filters: Optional[BaseFilter] = None,
+        **kwargs: Any,
+    ) -> List[Document]:
+        """Run similarity search with Firestore.
+        This method will throw if the index is not created, in which case you
+        will be prompted to create the index.
+
+        Raises:
+            FailedPrecondition: If the index is not created.
+
+        Args:
+            query: The query text.
+            k: The number of documents to return. Defaults to 4.
+            filters: The pre-filter to apply to the query. Defaults to None.
+
+        Returns:
+            List[Document]: List of documents most similar to the query text.
+        """
+
+        docs = self._similarity_search(
+            self.embedding.embed_query(query), k, filters=filters
+        )
+        return [
+            convert_firestore_document(
+                docs[i], page_content_fields=[self.content_field]
+            )
+            for i in range(len(docs))
+        ]
+
+    def similarity_search_by_vector(
```

Comment:
We are missing method: `similarity_search_with_score`

Reply:
This would require implementing a method to get distances and score the returned vectors from Firestore since the API doesn't return distances.